src/realm/object-store/sync/app.cpp

```diff
@@ -835,8 +835,12 @@ void App::log_in_with_credentials(const AppCredentials& credentials, const std::
                 return completion(nullptr,
                                   AppError(ErrorCodes::BadToken, "Could not log in user: received malformed JWT"));
             }
-            switch_user(user);
-            get_profile(user, std::move(completion));
+
+            get_profile(user, [this, completion = std::move(completion)](const std::shared_ptr<User>& user,
+                                                                         Optional<AppError> error) {
+                switch_user(user);
```

Comment:
We probably only want to switch users if the profile fetch succeeded?

Reply:
If the initial profile fetch fails then the user effectively isn't logged in yet and isn't really in a usable state.

Reply:
@tgoyne made the change, let me know if more is needed